sql/core/src/test/scala/org/apache/spark/sql/JsonFunctionsSuite.scala

```diff
@@ -390,11 +390,15 @@ class JsonFunctionsSuite extends QueryTest with SharedSparkSession {
 
   test("SPARK-24027: from_json of a map with unsupported key type") {
     val schema = MapType(StructType(StructField("f", IntegerType) :: Nil), StringType)
-
-    checkAnswer(Seq("""{{"f": 1}: "a"}""").toDS().select(from_json($"value", schema)),
-      Row(null))
```

Comment:
do you why this returns null before instead of a cast exception?

Reply:
@cloud-fan 
The difference is the type of the key
```
scala> val schema = MapType(StructType(StructField("f", IntegerType) :: Nil), StringType)
scala> Seq("""{{"f": 1}: "a"}""").toDS().select(from_json($"value", schema)).show
+-------+
|entries|
+-------+
|   null|
+-------+

scala> val schema2 = MapType(TimestampType, StringType)
scala> Seq("""{"2021-05-05T20:05:08": "a"}""").toDS().select(from_json($"value", schema2)).show

java.lang.ClassCastException: org.apache.spark.unsafe.types.UTF8String cannot be cast to java.lang.Long
  at scala.runtime.BoxesRunTime.unboxToLong(BoxesRunTime.java:107)
  at org.apache.spark.sql.catalyst.expressions.CastBase.$anonfun$castToString$8$adapted(Cast.scala:297)
  at org.apache.spark.sql.catalyst.expressions.CastBase.buildCast(Cast.scala:285)
  ...
```
I think the StructType in the key throws the exception in a different point and it's treated in a different way. Now it's homogeneous so an exception is thrown.

Reply:
Makes sense. Since it's not only improving error message, I'll merge this PR to master only.

Reply:
hm wait. I realised that `null` behaviour is same in `spark.json.read` too. Can we fix it together?

Reply:
Thanks @HyukjinKwon, you are right, I'm going to rise a new PR to fix this part.